trace/trace.go

```diff
@@ -215,21 +225,33 @@ func (s *Span) End() {
 	if !s.IsRecordingEvents() {
 		return
 	}
+
+	// If there are no exporters, no need to create spanData
+	// nor invoke all the code below this check, just return.
+	// See Issue 681.
+	if atomic.LoadInt64(&nExporters) <= 0 {
```

Comment:
Why not do this up-front when starting the span instead - avoid ever creating the SpanData object in the first place?